internal/app/machined/pkg/controllers/etcd/config.go

```diff
@@ -92,12 +92,27 @@ func (ctrl *ConfigController) Run(ctx context.Context, r controller.Runtime, log
 			return fmt.Errorf("error getting machine config: %w", err)
 		}
 
-		if err = safe.WriterModify(ctx, r, etcd.NewConfig(etcd.NamespaceName, etcd.ConfigID), func(status *etcd.Config) error {
-			status.TypedSpec().AdvertiseValidSubnets = machineConfig.Config().Cluster().Etcd().AdvertisedSubnets()
-			status.TypedSpec().ListenValidSubnets = machineConfig.Config().Cluster().Etcd().ListenSubnets()
+		if err = safe.WriterModify(ctx, r, etcd.NewConfig(etcd.NamespaceName, etcd.ConfigID), func(cfg *etcd.Config) error {
+			cfg.TypedSpec().AdvertiseValidSubnets = machineConfig.Config().Cluster().Etcd().AdvertisedSubnets()
+			cfg.TypedSpec().AdvertiseExcludeSubnets = nil
```

Comment:
excludesubnets already defined earlier?

Reply:
this makes sure we start with an empty list, otherwise `append()` below will keep appending more and more copies of the same VIP

Reply:
https://github.com/siderolabs/talos/blob/dce923f7471a73623c7bc99e376e51a53401faae/pkg/machinery/resources/etcd/config.go#L30 i was not sure if it was defined, checked and confirmed.